### PR TITLE
removed reliance on sessionStorage

### LINKED ISSUE #51: Remove sessionStorage from Variant widget
The variant plugin uses the config property 'variantDisplays' to read which properties to display. Currently, that's parsed by the plugin and stashed in window.sessionStorage. We really ought to be able to do something more elegant within the JB2 architecture. I dont know if any of these work, but some thoughts:

1) In NewTable(), which is ~line 86, do we have access to any of the jbrowse methods like getContainingTrack()? There must be something exposed within JB itself to let our code discover the owning track.

2) in selectFeature(), line 28, we can call getContainingTrack(). Assuming this would provide us with the variantDisplay data, can we pass this as a property to VariantWidget? I can understand some reluctance to track this onto featureData (although it's not crazy). Perhaps we add a typed property to VariantWidget:

const featureWidget = session.addWidget(
          'VariantWidget',
          widgetId,
          { featureData: feature.toJSON(), variantDisplayConfig: variantDisplay },
        )

### PR DESCRIPTION
**Pull request by jctrotter**

### CODE REVIEW

File: jbrowse/resources/external/minimalSession.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
           }
         }
       },
-      "variantDisplays": [
+      "metadata": [
```
Comment (bbimber) [inline jbrowse/resources/external/minimalSession.json]: OK, so this is an improvement, but isnt it a problem to pollute the top-level namespace like this? We should at least do:

metadata: {
   variantDisplayConfig: {....}
}

Note: I suggest we take this opportunity to rename the property to be more in line with our plugin too. Maybe 'extendedVariantDisplayConfig' would be even better?

File: jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantDisplay/model.js [hunk 2]
```diff
@@ -25,14 +25,16 @@ export default jbrowse => {
 
     .actions(self => ({
       selectFeature(feature) {
+        const widgetDisplayInfo = getContainingTrack(self).configuration.metadata.value
```
Comment (bbimber) [inline jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantDisplay/model.js]: and then just update this to look for the nested property (in a null-safe way)

File: jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/index.js [hunk 3]
```diff
@@ -10,14 +10,15 @@ export default jbrowse => {
       id: ElementId,
       type: types.literal('ExtendedVariantWidget'),
       featureData: types.frozen({}),
+      widgetDisplayInfo: types.frozen([])
```
Comment (bbimber) [inline jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/index.js]: For the sake of consistency, i'd rename this to match whatever we end up calling the property under track.metadata

**Review by bbimber — CHANGES_REQUESTED** [review]
overall seems like a good step forward. just a few general suggestions to keep it cleaner.

File: jbrowse/resources/external/minimalSession.json (hunk 1 above)
Comment (jctrotter) [reply jbrowse/resources/external/minimalSession.json]: Name and location within metadata updated.

File: jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantDisplay/model.js (hunk 2 above)
Comment (jctrotter) [reply jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantDisplay/model.js]: Updated to grab from new location in metadata. Null behavior handling maintained, can test by removing metadata from minimalSession and opening a track.

File: jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/index.js (hunk 3 above)
Comment (jctrotter) [reply jbrowse/src/client/JBrowse/Browser/plugins/ExtendedVariantPlugin/ExtendedVariantWidget/index.js]: Renamed.

**Pull request merged**